### PR TITLE
Revised String.Format method and its overloads

### LINKED ISSUE #3702: Basic revision of String.Format
The topic was revised to move documentation for all overloads onto what was then the overload list page. Now, all overloads are on a single page, and the format used to document them no longer is appropriate. Before beginning extensive revision, this page needs a basic reorganization to accommodate the current format.

### PR DESCRIPTION
**Pull request by rpetrusha**
# Revised String.Format method and its overloads 

## Summary

Revises **String.Format** so that the documentation is appropriate for the current page design, which includes all overloads on a single page.

Fixes #3702

## Details
In addition to deleting content made inappropriate by the design of docs.microsoft.com, I've done the following:
1. Recommended interpolated strings as an alternative to **String.Format**.
2. Moved examples so that they appear along with the appropriate overload.
3. Added minimal documentation for each overload. (The basic issue here is that you really don't have to pay any attention to which overload you're calling; the compiler resolves the call appropriately and transparently to the developer.)
4. Made C# examples runnable.
5. Added an **Examples** section with links to example sections.

## Suggested Reviewers

@mairaw @BillWagner

### CODE REVIEW

**Comment by rpetrusha** [comment]
Closing and reopening to begin new build.

**Comment by rpetrusha** [comment]
CLA stuck; closing and reopening.

**Comment by rpetrusha** [comment]
Closing and reopening to begin new build.

**Comment by rpetrusha** [comment]
The fixes to the build system do not seem to have addressed the build issue for String.Format. Closing and reopening to begin new build just in case.

**Comment by rpetrusha** [comment]
Closing and reopening to begin new build.

**Comment by TianqiZhang** [comment]
@rpetrusha build result is refreshed, please let me know if you are unblocked.

**Comment by rpetrusha** [comment]
The current content is now reflected in the build.

File: includes/provider-string-format.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+ 
+However, when calling the **String.Format** method, it is not necessary to focus on the particular overload that you want to call. Instead, you can call the method with an object that provides culture-sensitive or custom formatting and a [composite format string](~/docs/standard/base-types/composite-formatting.md) that includes one or more format items. You assign each format item a numeric index; the first index starts at 0. In addition to the initial string, your method call should have as many additional arguments as it has index values. For example, a string whose format items have indexes of 0 and 1 should have 2 arguments; one with indexes 0 through 5 should have 6 arguments. Your language compiler will then resolve your method call to a particular overload of the **String.Format** method.
```
Comment (BillWagner) [inline includes/provider-string-format.md]: There are performance considerations with respect to boxing here. If you call an overload of `string.Format` with a set of arguments of different types, that will likely resolve to the overload that takes a `params object[]`. If any of those types are value types, they will be boxed.

Should that be mentioned here and in the simple-string-format?

**Review by BillWagner — APPROVED** [review]
I'm going to approve this, but I think there are a couple other items we should address (either with new issues, or in this PR).  My one other comment is also optional to change.

These are all in the string.XML page:

1. The table for which version of the Unicode standard applies does not include .NET Core.
1. Many of the samples could use the REPL, if we choose. (Note sure we have enough data here yet though).

**Comment by rpetrusha** [comment]
@BillWagner, I'll add information on Unicode version support in .NET Core, and then merge the PR. I'll address the remaining comments in a separate PR. And I'll also investigate your comment about boxing and overload resolution -- my recollection, which could of course be wrong, is that Format(String, Object[]) is only called if there are four or more parameters; parameters of different types do not affect the overload called.

**Comment by BillWagner** [comment]
@rpetrusha I think something like this would involve boxing for the int:

```csharp
var times = 5;
var name = "Iago";
var s = string.Format("Hello {0}, I've seen you {1} times.", name, times);
```

Because of that, in a tight loop, one might write it as:

```csharp
var times = 5;
var name = "Iago";
var s = string.Format("Hello {0}, I've seen you {1} times.", name, times.ToString());
```

**Comment by rpetrusha** [comment]
@BillWagner, I've made the C# examples runnable and also added an **Examples** section with links to sections that have examples. I'll address the issues of Unicode support and boxing separately after I return from vacation.

**Pull request merged**